pkg/modules/meilisearch.go

```diff
@@ -107,34 +106,24 @@ func (m MeiliClient) GenerateCompetenceIndex(ctx context.Context, conn *bun.DB)
 	return nil
 }
 
-func getParentIds(conn *bun.DB, ctx context.Context, competenceID string) ([]string, error) {
-	var parents []db.Competence
-	err := conn.NewRaw("SELECT * FROM get_competence_tree_reverse(?)", competenceID).Scan(ctx, &parents)
-	if err != nil {
-		return nil, err
-	}
-
-	ids := make([]string, len(parents))
-	for i, parent := range parents {
-		ids[i] = parent.ID
-	}
-
-	return ids, nil
-}
-
-func getParentNames(conn *bun.DB, ctx context.Context, competenceID string) ([]string, error) {
-	var parents []db.Competence
-	err := conn.NewRaw("SELECT * FROM get_competence_tree_reverse(?)", competenceID).Scan(ctx, &parents)
-	if err != nil {
-		return nil, err
+func getParentDetails(competence db.Competence, competenceMap map[string]db.Competence) ([]string, []string) {
+	var parentIds []string
+	var parentNames []string
+
+	for competence.CompetenceID.Valid {
+		if parent, found := competenceMap[competence.CompetenceID.String]; found {
+			parentIds = append(parentIds, parent.ID)
+			parentNames = append(parentNames, parent.Name)
+			competence = parent
+		} else {
+			break
+		}
 	}
 
-	ids := make([]string, len(parents))
-	for i, parent := range parents {
-		ids[i] = parent.Name
-	}
+	slices.Reverse(parentIds)
+	slices.Reverse(parentNames)
 
-	return ids, nil
+	return parentIds, parentNames
 }
```

Comment:
Parents must be queried recursively. A skill has a parent which has a parent which has a parent and so on. Until the `CompetenceID` is nil. This seems to be missing. Am I overlooking something?

Reply:
Never mind, overlooked line 117 or smth